File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.py
def d(a = int):
    sum = arg = a;
    while arg > 0:
        sum += arg % 10;
        arg //= 10;
    return sum;
self = [True for i in range(10001)]
for i in range(10000):
    if d(i + 1) < 10000:
        self[d(i + 1)] = False
for i in range(1, 10000):
    if self[i]:
        print(i)

File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.cc
#include <iostream>
using namespace std;
int d(int n) {
	int tmp = n, sum = n;
	while (tmp > 0) {
		sum += tmp % 10;
		tmp /= 10;
	}
	return sum;
}
int main() {
	int i = 1;
	while (i <= 10000) {
		bool is_selfNum = true;
		for (int j = i; j > 0; j--) {
			if (i == d(j)) {
				is_selfNum = false;
				break;
			}
		}
		if (is_selfNum)
			cout << i << endl;
		i++;
	}
	return 0;
}